components/Form/index.js

```diff
@@ -0,0 +1,171 @@
+    }
+
+    setErrors(newErrors);
+    return isValid;
+  };
+
+  const formSubmitHandler = (event) => {
+    event.preventDefault();
+
+    if (validateForm()) {
+      const newProject = {
+        id: uuidv4(),
+        projectTitle: formData.title,
+        projectSize: formData.size,
+        projectDue: formData.dueBy,
+        tasks: formData.tasks.split(",").map((task) => task.trim()),
+      };
+
+      localStorage.setItem(
+        "projectsData",
+        JSON.stringify([...projectsData, newProject])
+      );
+
+      setProjectsData([...projectsData, newProject]);
+
+      setFormData({
+        title: "",
+        size: "",
+        dueBy: "",
+        tasks: "",
+      });
+
+      router.push("/");
+    }
+  };
+
+  const handleInputChange = (event) => {
+    const { name, value } = event.target;
+    setFormData((prevData) => ({
+      ...prevData,
+      [name]: value,
+    }));
+  };
+
+  return (
+    <>
+      <header className={styles.header}>
+        <h1 className={styles.h1}>Add new Project</h1>
+        <button
+          className={styles["navigate-btn"]}
+          onClick={() => router.push("/")}
+        >
+          Home
+        </button>
+      </header>
+      <form className={styles.form}>
+        <label>
+          Title
+          <input
+            className={styles.input}
+            type="text"
+            name="title"
+            value={formData.title}
+            onChange={handleInputChange}
+          />
+          {errors.title && <span className={styles.error}>{errors.title}</span>}
+        </label>
+        <label>
+          Size
+          <select
+            className={styles.select}
+            name="size"
+            value={formData.size}
+            onChange={handleInputChange}
+          >
+            <option value="" disabled>
+              Select Size
+            </option>
+            <option value="s">Small</option>
+            <option value="m">Medium</option>
+            <option value="l">Large</option>
+            <option value="xl">Extra Large</option>
+          </select>
+          {errors.size && <span className={styles.error}>{errors.size}</span>}
+        </label>
+        <label>
+          Due by
+          <input
+            className={styles.input}
+            type="date"
+            name="dueBy"
+            value={formData.dueBy}
+            onChange={handleInputChange}
+          />
+          {errors.dueBy && <span className={styles.error}>{errors.dueBy}</span>}
+        </label>
+        <label>
+          Tasks
+          <textarea
+            className={styles.inputarea}
+            type="text"
+            name="tasks"
+            value={formData.tasks}
+            onChange={handleInputChange}
+          />
+          {errors.tasks && <span className={styles.error}>{errors.tasks}</span>}
+        </label>
+        <button
+          type="submit"
+          className={styles.submit}
+          onClick={formSubmitHandler}
```

Comment:
The submit event is not triggered by the `onClick` of the submit button **but by the form.**
Add the `onSubmit` event to the form and move the `formSubmitHandler` up to the form.